pkg/connector/users.go

```diff
@@ -116,3 +183,114 @@ func getEmail(principal *v2.Resource) (string, error) {
+	}
+
+	o.usersMutex.RLock()
+	email, found := o.users[accountID]
+	o.usersMutex.RUnlock()
+
+	if !found {
+		return "", fmt.Errorf("user with account_id %s not found in cache", accountID)
+	}
+
+	return email, nil
+}
+
+// CreateAccountCapabilityDetails declares support for account provisioning without passwords.
+func (o *userBuilder) CreateAccountCapabilityDetails(ctx context.Context) (*v2.CredentialDetailsAccountProvisioning, annotations.Annotations, error) {
+	return &v2.CredentialDetailsAccountProvisioning{
+		SupportedCredentialOptions: []v2.CapabilityDetailCredentialOption{
+			v2.CapabilityDetailCredentialOption_CAPABILITY_DETAIL_CREDENTIAL_OPTION_NO_PASSWORD,
+		},
+		PreferredCredentialOption: v2.CapabilityDetailCredentialOption_CAPABILITY_DETAIL_CREDENTIAL_OPTION_NO_PASSWORD,
+	}, nil, nil
+}
+
+// CreateAccount provisions a new user in Dropbox Team based on AccountInfo.
+func (o *userBuilder) CreateAccount(
+	ctx context.Context,
+	accountInfo *v2.AccountInfo,
+	credentialOptions *v2.LocalCredentialOptions,
+) (
+	connectorbuilder.CreateAccountResponse,
+	[]*v2.PlaintextData,
+	annotations.Annotations,
+	error,
+) {
+	l := ctxzap.Extract(ctx)
+	profile := accountInfo.GetProfile().AsMap()
+
+	email, ok := profile["email"].(string)
+	if !ok || email == "" {
+		return nil, nil, nil, fmt.Errorf("email is required")
+	}
+
+	response, rateLimitData, err := o.AddMember(ctx, email)
+	var annos annotations.Annotations
+	annos.WithRateLimiting(rateLimitData)
+
+	if err != nil {
+		l.Error("error creating user", zap.Error(err))
+		return nil, nil, annos, err
+	}
+
+	if len(response.Complete) == 0 || response.Complete[0].Tag != "success" {
+		return nil, nil, annos, fmt.Errorf("failed to create user: unexpected response")
+	}
+
+	newUserProfile := response.Complete[0].Profile
+	userResource, err := userResource(newUserProfile, nil)
+	if err != nil {
+		l.Error("error converting created user to resource", zap.Error(err))
+		return nil, nil, annos, err
+	}
+
+	o.usersMutex.Lock()
+	if o.users != nil {
+		o.users[newUserProfile.AccountID] = newUserProfile.Email
+	}
+	o.usersMutex.Unlock()
+
+	return &v2.CreateAccountResponse_SuccessResult{
+		Resource: userResource,
+	}, []*v2.PlaintextData{}, annos, nil
+}
+
+// Delete implements account deprovisioning for users.
+func (o *userBuilder) Delete(ctx context.Context, resourceId *v2.ResourceId) (annotations.Annotations, error) {
+	l := ctxzap.Extract(ctx)
+
+	if resourceId.ResourceType != userResourceType.Id {
+		return nil, fmt.Errorf("invalid resource type: expected %s, got %s", userResourceType.Id, resourceId.ResourceType)
+	}
+
+	accountID := resourceId.Resource
+
+	userEmail, err := o.getEmailByAccountID(ctx, accountID)
+	if err != nil {
+		l.Error("error getting email for user", zap.Error(err), zap.String("accountID", accountID))
+		return nil, fmt.Errorf("failed to get email for account_id %s: %w", accountID, err)
+	}
+
+	_, rateLimitData, err := o.RemoveMember(ctx, userEmail)
```

Comment:
You can remove a member by their account id, we don't need this cache in that case no?

> Removes a member from a team. Permission : Team member management Exactly one of team_member_id, email, or external_id must be provided to identify the user account. 

https://www.dropbox.com/developers/documentation/http/teams#team-members-remove